### PR TITLE
ANW-1637: Fix GitHub Actions deprecation warnings

### PR DESCRIPTION
**Pull request by brianzelip**
### Before

https://github.com/archivesspace/archivesspace/actions/runs/3456487860

### After

https://github.com/archivesspace/archivesspace/actions/runs/3461105665

## Description

This PR addresses the deprecation warnings from all recent GitHub Actions workflows (eg see the Annotations summary here, https://github.com/archivesspace/archivesspace/actions/runs/3456487860):

1. Node.js 12 actions are deprecated. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/. Please update the following actions to use Node.js 16: actions/checkout, actions/setup-java, actions/setup-java, actions/checkout
2. The set-output command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/.

## Related JIRA Ticket or GitHub Issue

[ANW-1637](https://archivesspace.atlassian.net/browse/ANW-1637)

### Dependency upgrades

- relevant docs: [Supported version syntax](https://github.com/actions/setup-java#supported-version-syntax)
- relevant docs: [Supported distributions](https://github.com/actions/setup-java#supported-distributions)
- precedence: https://github.com/archivesspace/archivesspace/blob/v3.3.1/.github/workflows/release.yml#L12-L19
- tests successfully run and with no deprecation warnings herein, see eg [this PR's backend run](https://github.com/archivesspace/archivesspace/actions/runs/3461105665)

### `set-output`

- relevant docs: [Setting an output parameter](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter)
- manually logging variables to confirm values, see [Build Dist step here](https://github.com/archivesspace/archivesspace/actions/runs/3462846662/jobs/5782276182)

## ⚠️ What hasn't been tested ⚠️

- .github/workflows/release.yml

### CODE REVIEW

**Comment by brianzelip** [comment]
@quoideneuf  or @mark-cooper - what's the best way to test the Release workflow to make sure the new env output vars are working as expected?

**Comment by mark-cooper** [comment]
@brianzelip Nice detail in the description! LGTM. Minor merge conflict to resolve.

**Comment by brianzelip** [comment]
@mark-cooper conflict resolved

File: .github/workflows/frontend.yml [hunk 1]
```diff
@@ -38,6 +38,7 @@ jobs:
     - uses: actions/checkout@v3
     - uses: actions/setup-java@v3
       with:
+        distribution: 'temurin'
```
Comment (quoideneuf) [inline .github/workflows/frontend.yml:41]: @brianzelip Is this the case of us both doing the same thing independently? It looks like when this is merged we will have double `distribution:` values: yours in quotes, mine without. (No idea which is better or if it matters).

Comment (quoideneuf) [reply .github/workflows/frontend.yml:41]: ah, I understand now.

**Pull request closed without merging**